ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixParticipant.java

```diff
@@ -265,4 +273,18 @@ boolean setStoppedReplicas(List<String> stoppedReplicas) {
     instanceConfig.getRecord().setListField(ClusterMapUtils.STOPPED_REPLICAS_STR, stoppedReplicas);
     return helixAdmin.setInstanceConfig(clusterName, instanceName, instanceConfig);
   }
+
+  @Override
+  public void onPartitionLeadFromStandby(String partitionName) {
+    for (PartitionStateChangeListener partitionStateChangeListener : partitionStateChangeListeners) {
+      partitionStateChangeListener.onPartitionLeadFromStandby(partitionName);
```

Comment:
Thats true. As discussed offline, for enforcing ordering maybe we can define specific listeners (like storagelistener, replicationlistener) instead of defining a list of listeners. Can we consider this out of scope for this ticket?